Shorten the `src/core/crypto.js` file a little bit

Snuffleupagus opened the pull request.

Snuffleupagus committed:
Reduce duplication when checking the userPassword, in `CipherTransformFactory.prototype.#prepareKeyData`

Currently we duplicate the exact same code in both the `if`- and `else`-branches, which seems unnecessary, and we can also replace the manual loop.

Snuffleupagus committed:
Remove unnecessary `else if` when checking the encryptionKey in the `CipherTransformFactory` constructor

This can be simplified a tiny bit since we already throw `PasswordException` when no password is provided.

Snuffleupagus commented:
/botio-linux test

calixteman commented on src/core/crypto.js:
```diff
@@ -916,20 +916,16 @@ class CipherTransformFactory {
-        }
+    }
+
+    for (j = 0, n = checkData.length; j < n; ++j) {
```
nit: can't we use something like `return userPassword.every((x, i) => x === checkData[i]) ? encryptionKey : null;` ?

calixteman reviewed (approved):
LGTM. Thank you.

Snuffleupagus committed:
Change a couple of "password" ref-tests to "eq" tests

Currrently these are just "load" tests, and by also testing rendering we get slightly better test-coverage for the `src/core/crypto.js` file.

Snuffleupagus commented:
/botio unittest

Snuffleupagus commented:
/botio browsertest

Snuffleupagus commented:
/botio makeref
